analyzers/tests/SonarAnalyzer.UnitTest/TestCases/LoopsAndLinq.CSharp11.cs

```diff
@@ -7,13 +7,32 @@ class MyClass
     {
         void ListPattern(List<int[]> list)
         {
-            foreach (int[] array in list) // FN
+            foreach (int[] array in list) // Noncompliant
             {
-                if (array is [1, 2, 3]) // FN
+                if (array is [1, 2, 3]) // Secondary
                 {
-                    Console.WriteLine("Test");
+                    Console.WriteLine("Pattern match successful");
                 }
             }
+
+            foreach (var array in list) // Compliant, do not raise on var pattern in ListPattern
+            {
+                if (array is [1, var x, var z]) 
+                {
```

Comment:
Following up on the previous comment, if someone re-wrote this as a Linq Where, the `x` and `y` variables would not be properly scoped inside the `if` statement. Same with `is ... local`